x/commitment/keeper/msg_server_cancel_vest.go

```diff
@@ -51,6 +51,20 @@ func (k msgServer) CancelVest(goCtx context.Context, msg *types.MsgCancelVest) (
 		return nil, sdkerrors.Wrapf(types.ErrInsufficientVestingTokens, "denom: %s, amount: %s", msg.Denom, msg.Amount)
 	}
 
+	// Update the uncommitted tokens amount
+	uncommittedToken, found := commitments.GetUncommittedTokensForDenom(msg.Denom)
+
+	if found {
+		uncommittedToken.Amount = uncommittedToken.Amount.Add(msg.Amount)
+	} else {
+		uncommittedTokens := commitments.GetUncommittedTokens()
+		uncommittedTokens = append(uncommittedTokens, &types.UncommittedTokens{
+			Denom:  msg.Denom,
+			Amount: msg.Amount,
+		})
+		commitments.UncommittedTokens = uncommittedTokens
```

Comment:
@AustinoBombino I think this line need to apply regardless of the `found` condition correct?

Reply:
we basically need a way to set the changes applied at line 58 with `SetCommitments`

Reply:
@AustinoBombino I take it back `GetUncommittedTokensForDenom` is returning a pointer so updating it updates the sub item of `commitments`.